lib/cylc/conditional_simplifier.py

```diff
@@ -100,31 +101,40 @@ def nest_by_oper( self, nest_me, oper ):
         else:
             return nest_me
 
-    def clean_expr( self, nested_list, criteria ):
-        """Return a list with entries specified by 'critria' removed"""
+    def clean_expr( self, nested_list, criterion ):
+        """Return a list with entries specified by 'criterion' removed"""
         cleaned = copy.deepcopy( nested_list )
+
         # Make sure that we don't have extraneous nesting.
         while (isinstance(cleaned, list) and len(cleaned) == 1 and
                isinstance(cleaned[0], list)):
             cleaned = cleaned[0]
 
-        # Recurse through the nested list and remove criteria.
-        found = None
-        if isinstance(cleaned, str) or len(cleaned)==1:
-            if cleaned == criteria:
+        if len(cleaned) == 1:
+            cleaned = cleaned[0]
+
+        if isinstance(cleaned, str):
+            if cleaned == criterion:
                 return ""
             else:
                 return cleaned
+
+        # Recurse through the nested list and remove criterion.
+        found = None
         for i in range(0, len(cleaned)):
             if isinstance(cleaned[i], list):
-                cleaned[i] = self.clean_expr(cleaned[i], criteria)
-            if cleaned[i] == criteria:
+                cleaned[i] = self.clean_expr(cleaned[i], criterion)
+            if cleaned[i] in [criterion, '']:
```

Comment:
Single items in a nested expression get replaced by an empty string if cleaned up - these need to be removed at the higher level, e.g. `foo | ""` becomes just `foo`?